modules/registry/registry.go

```diff
@@ -203,3 +253,67 @@ func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustom
 
 	return c, nil
 }
+
+// SetDockerAuthConfig sets the DOCKER_AUTH_CONFIG environment variable with
+// authentication for with the given host, username and password.
+// It returns the base64 encoded credentials and a function to reset back
+// to the previous state.
+func SetDockerAuthConfig(host, username, password string, additional ...string) (creds string, cleanup func(), err error) { //nolint:nonamedreturns // Adds context to the return values.
+	creds, auth, err := DockerAuthConfig(host, username, password, additional...)
+	if err != nil {
+		return "", nil, fmt.Errorf("docker auth config: %w", err)
+	}
+
+	previousAuthConfig := os.Getenv("DOCKER_AUTH_CONFIG")
+	os.Setenv("DOCKER_AUTH_CONFIG", auth)
+
+	cleanup = func() {
+		if previousAuthConfig == "" {
+			os.Unsetenv("DOCKER_AUTH_CONFIG")
+			return
+		}
+		os.Setenv("DOCKER_AUTH_CONFIG", previousAuthConfig)
+	}
+
+	return creds, cleanup, nil
+}
+
+// DockerAuthConfig returns the base64 encoded credentials and the auth config
+// for the provided details. It also accepts additional host, username and password
+// triples to add more auth configurations.
+func DockerAuthConfig(host, username, password string, additional ...string) (creds, authConfig string, err error) { //nolint:nonamedreturns // Adds context to the return values.
```

Comment:
Just to understand this better, I guess this public function gets the auth config struct from the input data, right? It's useful for getting the "raw" config so that the tests can inject it in the test env (using `t.Setenv`). Also used by the module to set the env (using `os`).

Is this assumption correct?

Reply:
Correct it allows for `os` and `testing.T` flows.

My thinking was I didn't want to expose methods that took `testing.T` directly.

That said we could keep these private for now and see if there is a demand for them moving forward?

Reply:
I just realised I made it public because the tests are in `registry_test` package I could change that to just `registry`, what's the preferred cause?

Reply:
Thanks for the explanation 🙇 

If we make it private, then we need to change the test scope of the registry_test.go file, from `registry_test` to `registry`.  I'd prefer keeping the test scope so let's keep it public, and maybe a comment could be helpful. Wdyt about an internal package for this kind of things?

Reply:
That could work let me think on it.